Bug 1267483 : added file on configuring EFK

bfallonf opened the pull request:
@sosiouxme This is to do with BZ1267483:

https://bugzilla.redhat.com/show_bug.cgi?id=1267483

Pretty much just included the info from here:

https://github.com/openshift/origin-aggregated-logging

Question, though:

1. It's mentioned that it's a modified version of the ELK stack, but obviously I didn't want to use that abbreviation. Is it ok to call it EFK?

2. I've put in a section on running the deployer as a whole, and a section on running the individual parts (Kibana, Fluentd, etc.), but I'm worried I've confused the two. Did I just misunderstand that this is a thing the reader can do? When deploying the individual parts, is it still deployed by the template example I've put in? There doesn't seem to be anything that is being deployed... (I'm thinking something along the lines of the oo-install tool.)

3. The bits about the projects are vague. Does this need to be configured into a new project, then it applies to all projects? Or is it project specific, so when deploying it into a project, that project has the EFK stack?

4. I didn't include the Running the Deployer Script Locally bit, because I didn't understand it enough, but I think this ties into the downloading the deployer much like the installer tool, mentioned above. Could I ask for a little more info on this process, please?

I think that's all. I wouldn't recommend looking too much into what I have atm, as obviously I'm thinking it's not quite technically there just yet. I'll ask for a tech review in round 2. If this is too much I don't mind setting aside a time where we can chat over bluejeans or something. That might enable some more clearer understandings, and with the release impending, it might work better.

Thanks, much.

sosiouxme commented:
On Sun, Nov 1, 2015 at 10:46 PM, Brice Fallon-Freeman <
notifications@github.com> wrote:
>
>
>    1.
>
>    It's mentioned that it's a modified version of the ELK stack, but
>    obviously I didn't want to use that abbreviation. Is it ok to call it EFK?
>
>
It is OK to call it the EFK stack; however the reference to "ELK" is
something that will be a little familiar to some people, while "EFK" is not
in common usage, so it might be nice to provide that context.


>
>    1.
>
>    I've put in a section on running the deployer as a whole, and a
>    section on running the individual parts (Kibana, Fluentd, etc.), but I'm
>    worried I've confused the two. Did I just misunderstand that this is a
>    thing the reader can do? When deploying the individual parts, is it still
>    deployed by the template example I've put in? There doesn't seem to be
>    anything that is being deployed... (I'm thinking something along the lines
>    of the oo-install tool.)
>
>
The deployer actually defines everything needed for bootstrapping the
logging deployment; but manual intervention is required to complete the
deployment. I recognize the distinction is confusing. I would have liked to
just empower the deployer to do everything in one stroke but there were
security and sequencing reasons why it couldn't.

Running the deployer is a necessary prerequisite for the individual parts.
It creates a deployment for each of the ES instances, Kibana, and Fluentd.
You cannot separate these pieces from the deployer, and no part of the
stack is useful without the rest of it, so certainly they should not be
presented as independent instructions.

It might be helpful to consider it as a three-stage deployment:
1. Preparing to run the deployer
2. Running the deployer
3. Adjusting what was deployed


>
>    1.
>
>    The bits about the projects are vague. Does this need to be configured
>    into a new project, then it applies to all projects? Or is it project
>    specific, so when deploying it into a project, that project has the EFK
>    stack?
>
>
It is not project-specific. A single logging deployment collects logs for
all projects, for the entire OpenShift deployment. This is probably a good
thing to mention.

It doesn't have to be deployed into a new project, but it is strongly
suggested.


>
>    1.
>
>    I didn't include the Running the Deployer Script Locally bit, because
>    I didn't understand it enough, but I think this ties into the downloading
>    the deployer much like the installer tool, mentioned above. Could I ask for
>    a little more info on this process, please?
>
>
This is dev-only, no need to mention it. Folks that are trying to build
from the source github repos will need this, and they'll find it on github.


>
>
> I think that's all. I wouldn't recommend looking too much into what I have
> atm, as obviously I'm thinking it's not quite technically there just yet.
> I'll ask for a tech review in round 2. If this is too much I don't mind
> setting aside a time where we can chat over bluejeans or something. That
> might enable some more clearer understandings, and with the release
> impending, it might work better.
>

I've been adding a bit to the repo docs:
https://github.com/sosiouxme/origin-aggregated-logging/blob/20151028-docs/deployment/README.md
I was trying to expand/clarify some things, not sure how well I've done
that thus far.


> Thanks, much.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/openshift/openshift-docs/pull/1129
> Commit Summary
>
>    - Bug 1267483 : added file on configuring EFK
>
> File Changes
>
>    - *M* _build_cfg.yml
>    <https://github.com/openshift/openshift-docs/pull/1129/files#diff-0>
>    (2)
>    - *A* admin_guide/enabling_EFK.adoc
>    <https://github.com/openshift/openshift-docs/pull/1129/files#diff-1>
>    (541)
>
> Patch Links:
>
>    - https://github.com/openshift/openshift-docs/pull/1129.patch
>    - https://github.com/openshift/openshift-docs/pull/1129.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/1129>.
>

sosiouxme commented:
I guess I missed the most obvious thing I should say.

https://github.com/openshift/origin-aggregated-logging is the *repository* readme. It is for developers. It does not need to become docs.

https://github.com/openshift/origin-aggregated-logging/tree/master/deployment is directed at deployers. It should become real docs. (As noted, I have been working on expansion at https://github.com/sosiouxme/origin-aggregated-logging/blob/20151028-docs/deployment/README.md )

spadgett commented:
The asset config setting for logging must be set in master-config.yaml for the archived log button to appear in the web console. It must be an HTTPS URL.

```
assetConfig:
  ...
  loggingPublicURL: "https://<kibana-hostname>"
```

See https://github.com/openshift/openshift-docs/issues/1061

adellape commented:
@bfallonf I haven't read through your new topic or any of the comments here yet, but from a quick glance didn't notice mention of the existing https://github.com/openshift/openshift-docs/blob/master/admin_guide/aggregate_logging.adoc file. I was assuming that the existing topic would need to be either updated or removed as part of the new logging work, so just wanted to point it out here just in case.

bfallonf commented:
@sosiouxme OK. I think what I have now is in a better state. Thanks for the extra info. Can I get your thoughts? Also, it may be better to link out to the troubleshooting stuff? Or perhaps this would be a candidate for a troubleshooting docs (which are in the works). Big thanks for your help. 

@spadgett Is the details you've got here specific for the EFK stuff? It sounds like it might have a better home in another file, perhaps. (Which would tie into my question for Alex below...)

@adellape Are you suggesting this work be put into the aggregated logging file? That would sorta make sense. This is just another way to aggregate logging, right? In the file it mentions that more methods will be available soon...

adellape commented:
@bfallonf As I understand it (others please correct if wrong), this new work is the containerized and enhanced way of the same goal described in the old Aggregating Container Logs topic. So this new method should sunset the old for 3.1; I don't think we should have both topics exist. I wanted to mention it also in case there was any expository stuff in the old topic that get reworked into the new, but it doesn't look like there'd be much.

The `assetConfig` stuff @spadgett mentioned seems like it should go in the same place as your new work, as it would be a normal setup step for admins configuring logging, i.e., exposing the archived logging (that you just finished configuring) in the web console as well.

spadgett commented:
> @spadgett Is the details you've got here specific for the EFK stuff? It sounds like it might have a better home in another file, perhaps. (Which would tie into my question for Alex below...)

@bfallonf Yes, it is very specific to EFK. `loggingPublicURL` only works with this configuration.

spadgett commented:
Here's what it looks like in the web console by the way. The View Archive button links to Kibana.

![archive-button](https://cloud.githubusercontent.com/assets/1167259/10998317/51d75b30-8462-11e5-95b7-17d013b19e99.png)

adellape commented:
:fireworks:

bfallonf commented:
OK. I did the following:

1. Deleted the aggregated_logging file. There wasn't much to take over. Pretty much just changed a sentence in the overview. I also checked for any links that could break. There weren't any.

2. Added the loggingPublicURL parameter info into the Kibana section in the Post-Deployment section. I feel it's the most appropriate place. @spadgett can I get an ack this is the right info and in the right place?

3. Changed the title to something more substantial.

@sosiouxme @adellape @spadgett let me know if there's anything else. Thanks!

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+these settings to suit your own instance if desired. The template is found at
+the following GitHub repository:
+
+https://github.com/openshift/origin-aggregated-logging/blob/master/hack/templates/dev-builds.yaml
```
@sosiouxme Might this get shipped w/ one of the installer packages (atomic-openshift-utils or whatever is appropriate) at some point?

Trying to think of other options besides linking to GH, though I realize it's easier to update the template instead of updating the docs any time it changes.

@bfallonf We can link directly to a file in the doc source, which works fine on docs.openshift. The sticking point would be how the Portal build would behave (I think I've seen examples of that done in some Portal docs).

bfallonf replied:
@adellape Yes, I thought it'd be easier to link out rather than putting in the ginormous file into the docs. Maybe this led to me not fully understanding how the deployer is deployed. @sosiouxme open to ideas.

sosiouxme replied:
Yes, the idea is that it should point at a template ansible lays down in the enterprise case. The GH link would only be for the origin case. However it would be a different one anyway - this is a development template.

bfallonf replied:
Guys, sorry, I don't understand any of this. I'll see if I can talk to someone early tomorrow morning, my time.

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+* Fluentd: gathers logs from nodes and feeds them to Elasticsearch.
+* Kibana: a web UI for Elasticsearch.
+
+== Preperation for Deploying the EFK Stack
```
s/Preperation/Preparation/

adellape replied:
What about "Preparing to Deploy the EFK Stack"?

bfallonf replied:
I've decided I don't like either. Changed to "Pre-deployment Configuration"

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+Deploying the EFK stack requires:
+
+* A 'logging-deployer' secret
+* A 'logging-deployer' service account with the secret and create privileges
```
Mark up the secret and SA name instead of single quotes (probably just `*bold*`).

spadgett commented:
@adellape `loggingPublicURL` doc looks good. Might mention that it's the same hostname as `KIBANA_HOSTNAME` when calling `oc process`. Perhaps change the `loggingPublicURL` example to `kibana.example.com` used above?

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+
+== Overview
+
+Application developers can aggregate and view logs for a range of services using
```
This is a cluster admin task, but the intro makes it sound like "application developers" (OpenShift users) can set all this up. I think the intro should make the distinction more clear, and also give more context on what "a range of services"  means, and why a developer would be interested.

bfallonf replied:
Added a paragraph of context.

adellape commented:
> @adellape loggingPublicURL doc looks good. Might mention that it's the same hostname as KIBANA_HOSTNAME when calling oc process. Perhaps change the loggingPublicURL example to kibana.example.com used above?

@bfallonf ^ Mis-ping, I believe .

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+
+Deploy the entire EFK stack link:../dev_guide/templates.html[using a template],
+ensuring all future project configurations will feature the EFK stack. The
+default repository and branch of the template are set to origin/master. Change
```
@sosiouxme So I take it OSE admins need to update any of the parts pulling from origin on DockerHub to instead pull from the openshift3 repo on registry.access?

sosiouxme replied:
Yes, they should specify that in the IMAGE_PREFIX deployer parameter.

bfallonf replied:
I'm guessing this ties into the question above. I'll ask questions about that too.

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+
+[IMPORTANT]
+====
+The openshift-auth-proxy depends on a "node" base image, which is intended to be
```
The openshift-auth-proxy what? It's also mentioned suddenly without any context.

Looking at https://github.com/fabric8io/openshift-auth-proxy, maybe better to lead this with:

```
The OpenShift Authentication Proxy (*openshift-auth-proxy*) depends on a *node* base image [...]
```

sosiouxme replied:
There's no need to mention this anyway, as we don't need base images because we don't need to build the images, they're supplied. All the image building stuff is for people building from github.

bfallonf replied:
@sosiouxme So there's no need for this important box?

adellape replied:
Doesn't sound like it.

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+The openshift-auth-proxy depends on a "node" base image, which is intended to be
+the DockerHub nodejs base image. Defining standard templates also depends on a
+nodejs image called "node", which will be used instead of the intended image,
+causing the build to fail. To resolve this, delete the older node image:
```
For the above three lines:

s/nodejs/Node.js/
s/"node"/*node*/
s/node/*node*/

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+|A key to be used with the Ops Kibana certificate.
+
+|*_server-tls.json_*
+|JSON TLS options to override the Kibana server defaults. Refer to NodeJS docs for available options and the default options for an example.
```
s/NodeJS/Node.js/

sosiouxme replied:
The links from the original doc are needed for this description to be helpful...

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+
+|`*KIBANA_OPS_HOSTNAME*`, `*ES_OPS_INSTANCE_RAM*`, `*ES_OPS_CLUSTER_SIZE*`
+|Variables for the operations log cluster.
+|===
```
Lots of single and double quotes in the above table that should instead be properly marked up (CLI commands, image names, etc).

adellape commented on _build_cfg.yml:
```diff
@@ -210,6 +208,8 @@ Topics:
     File: syncing_groups_with_ldap
   - Name: Available Ports
     File: available_ports
+  - Name: Aggregating Logs with the EFK Stack
```
Can we drop "with the EFK Stack" in the title?

Also, the "Aggregating Container Logs" title was more specific as to what types of logs the topic is referring to, since there's other OpenShift service/platform logs that it could also be referring to.

sosiouxme replied:
Actually fluentd *does* feed service/platform logs into the system as well. I think taking over the old "aggregate_logging" filename would be fine.

adellape replied:
Sweet.

adellape commented on _build_cfg.yml:
```diff
@@ -210,6 +208,8 @@ Topics:
   - Name: Available Ports
     File: available_ports
+  - Name: Aggregating Logs with the EFK Stack
+    File: enabling_EFK
```
Whatever the topic title ends up being, the file name should probably be renamed to match it better.

adellape commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+= Aggregating Logs with the EFK Stack
```
Same comment as above wrt to the title, though you have more space in the topic title so "with the EFK Stack" may be fine to keep here.

sosiouxme commented:
>
>  Also, it may be better to link out to the troubleshooting stuff? Or
> perhaps this would be a candidate for a troubleshooting docs (which are in
> the works). Big thanks for your help.
>
I've given tips for troubleshooting the deployment itself at the time
you're performing it; these aren't things that will go wrong randomly later
on, so to me they don't seem like good candidates for general
troubleshooting docs, but are specific to this process.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+* A 'logging-deployer' secret
+* A 'logging-deployer' service account with the secret and create privileges
+* Sufficient volumes for ElasticSearch cluster storage
+* Optionally, a key and certificate for the Kibana server
```
These are part of the secret, don't think they need separate mention.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+The deployer generates the objects necessary for communication within the
+cluster and defines secrets and templates to implement aggregated logging.
+
+. Create a new project. Once implemented in a single project, the EFK stack will apply to every project within your OpenShift cluster:
```
"will apply to" would probably be clearer as "will collect logs for". The former makes it sound like each project has some kind of attribute set, which is not the case.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+|===
++
+You can supply the files when creating a new secret:
++
```
Should be clear the following is an example and not intended to be comprehensive.

bfallonf replied:
I do think this is clear enough. The values are certainly not going to work if they're pasted into any commands.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
++
+====
+----
+oc create -f - <<API
```
To be consistent, should start with a "$"

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+====
+----
+$ oc policy add-role-to-user edit \
+          system:serviceaccount:logging:logging-deployer
```
":logging:" is based on the project name. They might have used a different project name.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+Once the builds have successfully completed, the image streams will be populated
+and ready for deployment. However, you must first specify the `*IDEX_PREFIX*`
+parameter with the registry location.
+
```
Just skip all the above and fill out the deployer section.

bfallonf replied:
@sosiouxme Do you mean just the above paragraph? And do you mean with the info that you and @adellape were talking about above?

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
++
+====
+----
+oc process logging-deployer-template \
```
This is the template we expect the ansible installer to provide (although I think we'll need `-n openshift` to get it from the right namespace). For those with a non-ansible install (i.e. self-compiled origin) we should point at the source of the template (https://github.com/openshift/origin-aggregated-logging/blob/master/deployment/deployer.yaml).

bfallonf replied:
@sosiouxme So to clarify: This is ok for those who used the ansible installer, but we'll need to add in a -n. And separately, we'll have to add in another parameter pulling in that template for those that didn't?

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+====
++
+You can use the following environment variables when processing the kibana
+service:
```
They aren't environment variables (they're parameters) and you're not processing the kibana service, you're processing the deployer template.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+When setting the `*ENABLE_OPS_CLUSTER*` parameter to 'true', Fluentd splits logs
+between the ElasticSearch cluster and a cluster reserved for operations logs.
+This means a second ElasticSearch and Kibana are deployed. The deployments are
+distinguishable by the '-op' included in their names.
```
`-ops` not `-op`

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
++
+====
+----
+- system:serviceaccount:logging:aggregated-logging-fluentd
```
again `:logging:` comes from the project name, which could be different.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
++
+====
+----
+$ openshift admin policy add-cluster-role-to-user cluster-reader system:serviceaccount:logging:aggregated-logging-fluentd
```
likewise

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+=== ElasticSearch
+
+All pods in OpenShift deployments can safely share any volumes specified in the
+deployment. With ElasticSearch, all pods must have their own storage.
```
It's not very clear here that ElasticSearch usage is a contrast to the standard use case. And it needs to clear that the first sentence describes the standard use case, not an assurance of safety. It's a head-scratcher. Maybe:

"All pods from an OpenShift deployment share any volumes specified for the deployment. However, ElasticSearch pods cannot share storage."

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+This means a second ElasticSearch and Kibana are deployed. The deployments are
+distinguishable by the '-op' included in their names.
+
+. Next, as a cluster administrator, deploy the templates that were created by
```
It's *a* template actually. It does create several, but the user only deploys this one, and that's because it needs to be done as cluster-admin.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+----
+oc process logging-deployer-template \
+           -v KIBANA_HOSTNAME=kibana.example.com,PUBLIC_MASTER_URL=https://localhost:8443 \
+           | oc create -f -
```
Again the user should not literally cut and paste here; this is just an example. They should be choosing from the parameters listed above and filling in values according to their own needs.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+$ oc process logging-es-template | oc create -f -
+----
+====
+
```
Actually I wouldn't even mention scaling elasticsearch in these docs. It can maybe be done, but it would take a lot more detailed description to do with any kind of sanity. They should just create it with the intended scale from the get-go.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+----
+$ oc volume dc/logging-es-rca2m9u8 \
+          --add --overwrite --name=elasticsearch-storage \
+          --type=hostPath --path=/path/to/storage
```
This is just one possible example (and it's not an NFS volume, it's a volume on the host). That should be made clear. They can use any kind of volume they have available - there are lots. Do we have a generic reference for attaching volumes?

bfallonf replied:
I don't think we do have other examples... but I did neaten this up as suggested.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+*Node Selector*
+
+As ElasticSearch can use a lot resources, all members of a cluster should have
+low latency network connections to each other. Do this by directing the
```
"Do this" => "Ensure this"?

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+====
+----
+$ oc scale dc/logging-fluentd --replicas=3
+$ oc scale rc/logging-fluentd-1 --replicas=3
```
Here "3" should be substituted with the number of nodes in OpenShift. And it should be noted that this needs to be adjusted if nodes are added or subtracted.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+====
+
+Setting the `loggingPublicURL` parameter creates a *View Archive* button on the
+OpenShift web console under the *Browse* tab, then finding the desired pod. This
```
", then finding the desired pod" doesn't make sense.

sosiouxme commented on admin_guide/enabling_EFK.adoc:
```diff
@@ -0,0 +1,357 @@
+====
+
+You can see the UI by visiting the site specified at the '*KIBANA_HOSTNAME*'
+variable.
```
... assuming you have a router and DNS points at it appropriately for this address. Which we should probably mention much earlier.

bfallonf replied:
Didn't see that in your updated doc. Added to the list.

bfallonf commented:
@spadgett @adellape @sosiouxme Thanks for the info. I covered most of the typo/suggestion stuff, but I don't understand some of the things brought up by Alex... I'll see if I can chat tomorrow morning (my time) with someone.

bfallonf commented:
@adellape @sosiouxme Alex, I've updated the problem bits after our talk this morning (yesterday, for you). 

There's a few unanswered questions for Luke above. I'm not sure of the urgency of them, but at least it's one step further to completion. 

If there's anything else, let me know.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,475 @@ toc::[]
 ====
 
-[[creating-logging-pods]]
+You can any volume type, such as
```
"You can any"

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,475 @@ toc::[]
 
-[[creating-logging-pods]]
+You can any volume type, such as
+link:../admin_guide/persistent_storage_nfs.html[NFS].
```
This file got moved, link should be `../install_config/persistent_storage/persistent_storage_nfs.html`

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,475 @@ toc::[]
-<1> The name for the master that will be used during logging.
-<2> The IP or a DNS resolvable name used to access the master.
-====
+$ oc create -f https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/infrastructure-templates/enterprise/logging-deployer.yaml
```
For OSE, it's installed by default by the installation utility at `/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/infrastructure-templates/enterprise/logging-deployer.yaml`. So the command for OSE should be:

```
$ oc create -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/infrastructure-templates/enterprise/logging-deployer.yaml
```

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,475 @@ toc::[]
 ----
-# systemctl enable td-agent
-# systemctl start td-agent
+oc process logging-deployer-template -n openshift \
```
Missing `$` command prompt.

adellape commented:
@bfallonf Comments inline.

If @sosiouxme doesn't have any more immediate comments, I'm OK with merging this after the above are addressed and we can iterate.

adellape commented:
Also squash when done.

chunyunchen commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
-Add the new service account to the `*users*` list at the end of the SCC object
-(*default* is the project name):
+To access the Kibana console from the OpenShift web console, add the
+`loggingPublicURL` parameter to the *_/etc/openshift/master/master-config.yaml_*
```
For OSE/AEP, the file master-config.yaml is under /etc/origin/master/.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
+cluster and defines secrets and templates to implement aggregated logging.
 
-. Run the following commands:
+. Create a new project. Once implemented in a single project, the EFK stack will
```
Since later steps say stuff like "for example, if your project was named `logging`", it would be nice if the example precedent was set at this project creation step.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
-. Create the *_/etc/td-agent/config.d/kubernetes.conf_* file with the following
-contents:
++
+Then, ensure the logging-deployer can edit the service. The following example
```
Mark up `*logging-deployer*`.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
-        </server>
+The EFK stack is deployed link:../dev_guide/templates.html[using a template].
+This ensures all future project configurations will feature the EFK stack. The
+default repository and branch of the template are set to origin/master. Change
```
"The default repository and branch of the template are set to origin/master." Is this still true for OSE?

bfallonf replied:
@adellape @sosiouxme I'm not too sure if I understand the question... I tried looking in the enterprise and origin template and couldn't find the answer.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
-# chown td-agent:td-agent /etc/td-agent/config.d
+$ oc process logging-deployer-template -n openshift \
+           -v KIBANA_HOSTNAME=kibana.example.com,PUBLIC_MASTER_URL=https://localhost:8443 \
+           | oc create -f -
```
It looks like `ES_CLUSTER_SIZE` is also required:

```
# oc process logging-deployer-template             -v KIBANA_HOSTNAME=adellape-111115-quick.os1.phx2.redhat.com,PUBLIC_MASTER_URL=https://adellape-111115-quick.os1.phx2.redhat.com:8443            | oc create -f -
error processing the template "logging-deployer-template": template "logging-deployer-template" is invalid: parameters: invalid value '{Name:ES_CLUSTER_SIZE DisplayName: Description:How many instances of ElasticSearch to deploy. Value: Generate: From: Required:true}', Details: template.parameters[8]: parameter ES_CLUSTER_SIZE is required and must be specified
error: no objects passed to create
```

I set it to 1 and it created fine.

adellape commented on install_config/aggregate_logging.adoc:
```diff
@@ -12,628 +12,482 @@ toc::[]
 ====
-Any errors are logged in the *_/var/log/td-agent/td-agent.log_* file.
+----
+$ openshift admin policy add-cluster-role-to-user cluster-reader system:serviceaccount:logging:aggregated-logging-fluentd
```
Use "oadm" instead of "openshift admin".

bfallonf commented:
@adellape @chunyunchen Thanks for the comments. Everything taken care of except for the one by Alex i didn't fully understand...

Let me know if there's anything else.
